### PR TITLE
refactor: implement sub-commands and validate container image

### PR DESCRIPTION
**Pull request by amannocci**
## What is the change being made?
* Implement sub-command `run` and `version` for `apmsoak` using [cobra](https://cobra.dev/).
* Rename `Dockerfile` into the standard `Containerfile`.
* Extract CI steps in Makefile targets to be able to reproduce it locally.
* Add `.go-version` file with the current go version.
* Add standard labels on the container image.
* Add env variable support for all flags.
* Move non-public code to the internal package.

## Why is the change being made?
* Add the ability to validate the container image without dependencies before pushing it to the registry.

## How has this been tested?
* Run `make build` to build the binary.
* Run the following command with a serverless project

```bash
ELASTIC_APM_SERVER_URL="https://<project_id>.<url>" ELASTIC_APM_API_KEYS="<project_id>:<api_key>" ./dist/apmsoak run -f cmd/apmsoak/scenarios_test.yml
```

### CODE REVIEW

File: Containerfile
Comment (axw) [inline Containerfile:1]: TIL Containerfile is a thing :)

**Review by axw — APPROVED** [review]
Nice, LGTM!

**Comment by axw** [comment]
Sorry for the delay @amannocci, this got buried in my inbox.

**Comment by dmathieu** [comment]
The move to internal packages here is preventing apm-server from using the latest versions of apm-perf, since it needs the loadgen package for its system tests.
https://github.com/elastic/apm-server/blob/821d5f5b05c42da6d512d53166ea3ebfdfad977c/systemtest/soaktest/soaktest.go#L30

**Comment by axw** [comment]
@dmathieu we should be replacing the apmsoak in apm-server with the one in apm-perf

**Pull request merged**